### DISCUSSION TITLE
[2기 -A] 장치근

### DISCUSSION THREAD

**Discussion by wkdclrms123** [super-corini/programmers_kdt_II#354] [2기 Monthly Project I]
## EDA Monthly Project
- 웹사이트 주소 : http://ec2-54-180-28-48.ap-northeast-2.compute.amazonaws.com:5000/
- github 주소 : https://github.com/wkdclrms123/Monthly_Proj.git

<hr>

### 프로젝트 진행
- 1. 작업중 github에 commit과 push를 안해서그런지는 모르겠지만 작업을 열심히 하고 다음날 컴퓨터를 켜보니 모든 작업물이 없어져있고 venv파일만 쓸쓸하게 남아있는 안타까움을 겪었습니다..
- 2. 웹상 그래프 업로드를 위해 계획했던 것이 이미지파일을 그냥 올리는 것이었는데 이번 Pre-Monthly Proj에서 Plotly라는 것을 알게되어 굉장히 도움이 되었습니다.
- 3. 서버에 올라간 파일을 수정할 때 파일을 삭제한 후 다시 clone작업을 했는데 `rm -rf Monthly_Proj` 를 사용할때마다 손이 벌벌떨렸습니다...
- 4. requirements.txt 작성 후 서버 내에서 `pip install -r requirements.txt` 명령어를 사용하는 중간중간 모듈의 버전이 너무 높아 설치가 안되는 오류가 발생했습니다.
        - 해결방법 : github에 있는 requirements.txt의 버전을 일일이 수정해 주었습니다.
- 5. 서버를 오픈하는 방법에 대해 거의 몰랐는데 이번에 aws를 열기 위해 **~~3억건의~~ 구글링**을 통해 많이 배웠습니다.
- 6. 사실 지금도 aws 시간초과되어 과금되는것이아닌가.. 에 대한 걱정이 너무나 듭니다... ㅠㅠ
        - 21/5/24/18:24 해결완료
        - 멍청하게 인스턴스를 t2-large로 설정해버려서 이미 4300원가량 과금되었음.. 
        - 다시 인스턴스를 t2-micro로 설정한 후 다시 서버오픈
- 7. 전체화면을 하지않으면 그래프가 튀어나가버리는 현상이 일어나버려서 전체화면을 부탁드립니다.
<hr>

**Comment by super-corini** [comment]
- t2.micro의 경우에만 Free Tier가 적용됩니다 😢 
- rm -rf의 경우에는 신중해야죠...
- 한번 AWS EC2를 E2E로 해보시면 그 이후엔 다른 GCP, Azure 도 금방 사용하실 수 있을거에요 👍 
수고하셨습니당 😊
